zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/AuthorizationRequestValidator.java

```diff
@@ -31,6 +32,33 @@ public final class AuthorizationRequestValidator {
               PermissionTypeEnum.DELETE_DRD, ResourceTypeEnum.DEPLOYMENT,
               PermissionTypeEnum.DELETE_FORM, ResourceTypeEnum.DEPLOYMENT);
 
+  public static Optional<ProblemDetail> validateAuthorizationCreateRequest(
+      final AuthorizationCreateRequest request) {
+    return validate(
+        violations -> {
+          // owner validation
+          if (request.getOwnerId() == null) {
+            violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("ownerId"));
+          }
+          if (request.getOwnerType() == null) {
+            violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("ownerType"));
+          }
+
+          // resource validation
+          if (request.getResourceId() == null) {
+            violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("resourceId"));
+          }
+          if (request.getResourceType() == null) {
+            violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("resourceType"));
+          }
```

Comment:
❓ Should we also check for empty Strings in these properties?

Reply:
Yes, we should 👍